google/auth/action.yml

```diff
@@ -0,0 +1,21 @@
+name: google/auth
+
+description: |
+  This is an opinionated GitHub Action to authenticate with GCP.
+  It generates a Workload Identity Pool Provider ID based on the repository name, which is compatible with the
+  GCP Workload Identity Pool Provider ID we use for Elastic Observability repositories.
+
+runs:
+  using: composite
+  steps:
+    - name: Generate workloadIdentityPool provider ID
+      id: generate-workload-identity-pool-provider-id
+      run: |
+        HASH=$(echo -n "${GITHUB_REPOSITORY}" | sha256sum | cut -c1-27)
+        echo "workload_identity_provider_id=repo-${HASH}" >> "${GITHUB_OUTPUT}"
+      shell: bash
+    - uses: google-github-actions/auth@v2
```

Comment:
Do we want to pin versions?

Reply:
https://github.com/elastic/oblt-actions/pull/3/commits/4ea3a44a9fe36e5d20395f6cbf877f7176b95e89